File: java/src/main/java/shapes/client/shell/Handler.java
package shapes.client.shell;

import static shapes.client.shell.InputReader.readNumber;

import shapes.base.models.Circle;
import shapes.base.models.Rectangle;
import shapes.base.models.Square;

final class Handler {
    static void handleCircle() {
        System.out.println("What is the radius of the circle?");
        double radius = readNumber();
        Circle circle = new Circle(radius);
        System.out.printf("Area: %.3f%n", circle.area());
        System.out.printf("Perimeter: %.3f%n", circle.perimeter());
    }

    static void handleRectangle() {
        System.out.println("What is the width of the rectangle?");
        double width = readNumber();
        System.out.println("What is the height of the rectangle?");
        double length = readNumber();
        Rectangle rectangle = new Rectangle(length, width);
        System.out.printf("Area: %.3f%n", rectangle.area());
        System.out.printf("Perimeter: %.3f%n", rectangle.perimeter());
    }

    static void handleSquare() {
        System.out.println("What is the side length of the square?");
        double side = readNumber();
        Square square = new Square(side);
        System.out.printf("Area: %.3f%n", square.area());
        System.out.printf("Perimeter: %.3f%n", square.perimeter());
    }
}

File: csharp/Src/Shapes/Client/Shell/Handler.cs
using Shapes.Base.Models;

using static Shapes.Client.Shell.InputReader;

namespace Shapes.Client.Shell;

sealed class Handler {
    public static void HandleCircle() {
        Console.WriteLine("What is the radius of the circle?");
        double radius = ReadNumber();
        Circle circle = new Circle(radius);
        Console.WriteLine("Area: {0:F3}", circle.Area());
        Console.WriteLine("Circumference: {0:F3}", circle.Perimeter());
    }

    public static void HandleRectangle() {
        Console.WriteLine("What is the width of the rectangle?");
        double width = ReadNumber();
        Console.WriteLine("What is the height of the rectangle?");
        double height = ReadNumber();
        Rectangle rectangle = new Rectangle(width, height);
        Console.WriteLine("Area: {0:F3}", rectangle.Area());
        Console.WriteLine("Perimeter: {0:F3}", rectangle.Perimeter());
    }

    public static void HandleSquare() {
        Console.WriteLine("What is the side length of the square?");
        double side = ReadNumber();
        Square square = new Square(side);
        Console.WriteLine("Area: {0:F3}", square.Area());
        Console.WriteLine("Perimeter: {0:F3}", square.Perimeter());
    }
}
